pkg/executor/wasm/spec.go

```diff
@@ -0,0 +1,107 @@
+package wasm
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+)
+
+const (
+	EngineType                    = "wasm"
+	EngineKeyEntryModule          = "entrymodule"
+	EngineKeyEntrypoint           = "entrypoint"
+	EngineKeyParameters           = "parameters"
+	EngineKeyEnvironmentVariables = "environmentvariables"
+	EngineKeyImportModules        = "importmodules"
+)
+
+func NewEngineSpec(
+	entryModule model.StorageSpec,
+	entrypoint string,
+	parameters []string,
+	environmentVariables map[string]string,
+	importModules []model.StorageSpec) model.EngineSpec {
+	return model.EngineSpec{
+		Type: EngineType,
+		Params: map[string]interface{}{
+			EngineKeyEntryModule:          entryModule,
+			EngineKeyEntrypoint:           entrypoint,
+			EngineKeyParameters:           parameters,
+			EngineKeyEnvironmentVariables: environmentVariables,
+			EngineKeyImportModules:        importModules,
+		},
+	}
+}
+
+// Describes a raw WASM job
+type Engine struct {
```

Comment:
Why do we have this in both `model` and here?

Reply:
Eventually these specifications will live inside plugins - I'll need to figure our a way of exposing their "schemas" to client and requester nodes, but punting that task for now. 
I'll dedup these specs to a single location (probably `pkg/model`) before merging.